### PR TITLE
add short example program; expand README with example installation

### PR DESCRIPTION
**Pull request by conradsnicta**
This is to partially address the concerns raised by JMLR reviewer 1.
Specifically, the short example program has been added to satisfy the following comment:

> The documentation of ensmallen is detailed but I missed some simple examples in the README file. There are examples in the documentation, but I believe having a few compileable-then-runnable examples in the repository would be a great improvement.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,6 +28,20 @@ If CMake is not already available on your system, it can be obtained from https:
...
+mkdir build
+cd build
+cmake ..
+sudo make install
+```
+
+
+### Example Usage
+
+See `example.cpp` for example usage of the L-BFGS optimizer.
```
Comment (coatless) [inline README.md:43]: Maybe add a link to the example and briefly describe the setting?

```suggestion
See [`example.cpp`](example.cpp) for example usage of the L-BFGS optimizer in a linear regression setting.
```

File: example.cpp [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  for (size_t i = 0; i < n_points; ++i)
+  {
+    double a = arma::randu();
+    X(1, i) += a;
+    y(i)    += a;
+  }
+  
+  LinearRegressionFunction lrf(X, y);
+  
+  ens::L_BFGS opt;
```
Comment (zoq) [inline example.cpp:57]:
```suggestion
  // create an Limited-memory BFGS optimizer object with default parameters
  ens::L_BFGS opt;
```

Just to provide some additional information.

File: example.cpp [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  opt.MaxIterations() = 10;
+  
+  arma::vec theta(n_dims, arma::fill::randu);   // initial point (uniform random)
```
Comment (zoq) [inline example.cpp:60]:
```suggestion
  // initial point (uniform random)
  arma::vec theta(n_dims, arma::fill::randu);
```

Just to be consistent with the comment style.

File: README.md (hunk 1 above)
Comment (conradsnicta) [reply README.md]: thanks :)

File: example.cpp (hunk 2 above)
Comment (conradsnicta) [reply example.cpp:58]: good idea

**Review by zoq — APPROVED** [review]
Looks good to me.

File: example.cpp [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  for (size_t i = 0; i < n_points; ++i)
+  {
+    double a = arma::randu();
+    X(1, i) += a;
+    y(i)    += a;
+  }
+  
+  LinearRegressionFunction lrf(X, y);
+  
+  // create an Limited-memory BFGS optimizer object with default parameters
```
Comment (rcurtin) [inline example.cpp:57]:
```suggestion
  // create a Limited-memory BFGS optimizer object with default parameters
```

File: example.cpp [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  
+  // initial point (uniform random)
+  arma::vec theta(n_dims, arma::fill::randu);
+  
+  opt.Optimize(lrf, theta);
+  
+  // theta now contains the optimized parameters
+  theta.print("theta:");
+  
+  return 0;
```
Comment (rcurtin) [inline example.cpp:69]:
```suggestion
```
Depending on your level of pedantry: technically this isn't necessary in C++.  Up to you whether you want to remove it---it doesn't make a difference to me.

**Review by rcurtin — APPROVED** [review]
Awesome, thanks @conradsnicta for adding this!  I think it might be useful to also add a link in the documentation, so that when a user goes to http://ensmallen.org/docs.html they can find this example too.  I'm not sure what the best way to do that is, but honestly just a link to the Github file would be sufficient, I think.  Once we merge this, I'll open a quick PR adding that link.

**Comment by rcurtin** [comment]
I added a link to the file on ensmallen.org. :+1:

**Pull request merged**